### PR TITLE
Added support for Sony ODA as a backend

### PR DESCRIPTION
**Pull request by Click2cloud-Pallas**
This will allow the following operations with Sony ODA as Object Storage Backend:
1. Create Backend
2. Delete Backend
3. Create s3 Bucket (work in progress)
4. Delete s3 Bucket (work in progress)
5. Upload Object (work in progress)
6. Download Object (work in progress)
7. Delete Object (work in progress)
8. cross-cloud migration as destination (work in progress)

### CODE REVIEW

**Comment by Click2Cloud-SGupta** [comment]
Sony ODA added as backend

**Comment by Click2cloud-Pallas** [comment]
This update will allow the following operations with Sony ODA as Object Storage Backend:

Create s3 Bucket 
Upload Object
Download Object 
cross-cloud migration as destination

**Comment by Click2cloud-Pallas** [comment]
Hi, @kumarashit please review , Thanks.

File: backend/pkg/utils/constants/constants.go [hunk 1]
```diff
@@ -30,4 +30,5 @@ const (
 	BackendTypeYIGS3     = "yig"
 	BackendTypeAlibaba   = "alibaba-oss"
 	BackendTypeHpcBlock  = "hw-block"
+	BackendTypeSonyODA   = "sony-s3"
```
Comment (kumarashit) [inline backend/pkg/utils/constants/constants.go:33]: Can we call it sony-oda

File: dataflow/pkg/model/dataflow_type.go [hunk 2]
```diff
@@ -33,6 +33,7 @@ var (
 	STOR_TYPE_HW_FUSIONCLOUD   = "hw-fusioncloud"
 	STOR_TYPE_IBM_COS          = "ibm-cos"
 	STOR_TYPE_ALIBABA_OSS      = "alibaba-oss"
+	STOR_TYPE_SONY_ODA         = "sony-s3"
```
Comment (kumarashit) [inline dataflow/pkg/model/dataflow_type.go:36]: Same here

File: datamover/pkg/sonyoda/odalifecycle.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	. "github.com/opensds/multi-cloud/datamover/pkg/utils"
+	log "github.com/sirupsen/logrus"
+)
+
+//Sony-ODA only support standard(Tier-1)
```
Comment (kumarashit) [inline datamover/pkg/sonyoda/odalifecycle.go:12]: if this is not required/applicable then please remove it. SONY ODA doesn't support different storage class.

File: s3/pkg/datastore/aws/aws.go [hunk 4]
```diff
@@ -43,26 +43,26 @@ import (
 )
 
 type AwsAdapter struct {
-	backend *backendpb.BackendDetail
-	session *session.Session
+	Backend *backendpb.BackendDetail
```
Comment (kumarashit) [inline s3/pkg/datastore/aws/aws.go:46]: As these are not SONY ODA specific changes, these fmt changes can be done as part of separate optimization PR

**Review by kumarashit — COMMENTED** [review]
These cover the use case of policy based lifecycle as well as migration but didn't find PUT/GET/DELETE/POST for SONY ODA?

**Comment by Click2cloud-Pallas** [comment]
> These cover the use case of policy based lifecycle as well as migration but didn't find PUT/GET/DELETE/POST for SONY ODA?

Yes, Because we are redirecting it to AWS SDKs.

**Comment by kumarashit** [comment]
@Click2cloud-Pallas When this PR gets merged, you need to open another to sync to master

**Review by kumarashit — APPROVED** [review]
LGTM

**Review by NajmudheenCT — APPROVED** [review]
LGTM

**Pull request merged**